content/en/blog/_posts/2025-04-23-Kubernetes-v1-33-Release/index.md

```diff
@@ -0,0 +1,453 @@
+
+{{< figure src="k8s-1.33.png" alt="Kubernetes v1.33 logo: XX" class="release-logo" >}}
+
+## Spotlight on key updates
+
+Kubernetes v1.33 is packed with new features and improvements. Here are a few select updates the Release team would like to highlight!
+
+### Stable: Sidecar containers
+
+The sidecar pattern involves deploying separate auxiliary container(s) to handle extra capabilities in areas such as networking, logging, and metrics gathering. Sidecar containers graduate to stable in v1.33.
+
+Kubernetes implements sidecars as a special class of init containers with `restartPolicy: Always`, ensuring that sidecars start before application containers, remain running throughout the pod's lifecycle, and terminate automatically after the main containers exit.
+
+Additionally, sidecars can utilize probes (startup, readiness, liveness) to signal their operational state, and their Out-Of-Memory (OOM) score adjustments are aligned with primary containers to prevent premature termination under memory pressure.
+
+To learn more, read [Sidecar Containers](/docs/concepts/workloads/pods/sidecar-containers/).
+
+This work was done as part of [KEP-753: Sidecar Containers](https://kep.k8s.io/753) led by SIG Node.
+
+### Beta: In-place resource resize for vertical scaling of Pods
+
+When provisioning a Pod, you can use various resources such as Deployment, StatefulSet, etc. Scalability requirements may need horizontal scaling by updating the Pod replica count, or vertical scaling by updating resources allocated to Pod’s container(s). Before this enhancement, container resources defined in a Pod's `spec` were immutable, and updating any of these details within a Pod template would trigger Pod replacement.
+
+But what if you could dynamically update the resource configuration for your existing Pods without restarting them?
+
+The [KEP-1287](https://kep.k8s.io/1287) is precisely to allow such in-place Pod updates. It was released as alpha in v1.27, and has graduated to beta in v1.33. This opens up various possibilities of vertical scale-up for stateful processes without any downtime,
+seamless scale-down when the traffic is low, and even allocating larger resources during startup,
+which can then be reduced once the initial setup is complete.
+
+This work was done as part of [KEP-1287: In-Place Update of Pod Resources](https://kep.k8s.io/1287) led by SIG Node and SIG Autoscaling.
+
+### Alpha: New configuration option for kubectl with `.kuberc` for user preferences
+
+`kubectl` introduces an entirely new opt-in configuration file `.kuberc` for user preferences, allowing users to manage cluster credentials and host information separately. During the alpha stage, users can enable this by setting the environment variable of `KUBECTL_KUBERC=true` set, along with the `.kuberc` configuration file, which is under `~/.kube/kuberc` by default. This can also be run by using the argument of `--kuberc`, for example: `kubectl --kuberc /var/kube/rc`.
```

Comment:
```suggestion
`kubectl` introduces an entirely new opt-in configuration file `.kuberc` for user preferences, allowing users to manage cluster credentials and host information separately. During the alpha stage, users can enable this by setting the environment variable of `KUBECTL_KUBERC=true`, along with the `.kuberc` configuration file, which is under `~/.kube/kuberc` by default. This can also be run by using the argument of `--kuberc`, for example: `kubectl --kuberc /var/kube/rc`.
```